projects/storefrontlib/src/layout/theme/site-theme.service.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  ComponentRef,
+  Injectable,
+  Renderer2,
+  RendererFactory2,
+} from '@angular/core';
+import { SiteContextConfig, THEME_CONTEXT_ID } from '@spartacus/core';
+
+@Injectable({ providedIn: 'root' })
+export class SiteThemeService {
+  protected rootComponent: ComponentRef<any>;
+  protected renderer: Renderer2;
+  protected existingTheme: string;
+
+  constructor(
+    protected config: SiteContextConfig,
+    protected rendererFactory: RendererFactory2
+  ) {}
+
+  init(rootComponent: ComponentRef<any>): void {
```

Comment:
Can we add a bit of explanation here to help the consumer of this service the reason why we have an init method? This helps to understands that this should only be applied once.